src/BlazorWasmSdk/Tasks/ComputeBlazorPublishAssets.cs

```diff
@@ -549,24 +552,58 @@ private void ApplyPublishProperties(ITaskItem newAsemblyAsset)
                     {
                         var finalCulture = !string.IsNullOrEmpty(culture) ? culture : inferredCulture;
                         var assemblyName = Path.GetFileName(candidate.GetMetadata("RelativePath").Replace("\\", "/"));
-                        satelliteAssemblyToPublish.Add((finalCulture, assemblyName), candidate);
+                        if (!satelliteAssemblyToPublish.ContainsKey((finalCulture, assemblyName)))
+                        {
+                            satelliteAssemblyToPublish.Add((finalCulture, assemblyName), candidate);
+                        }
+                        else
+                        {
+                            Log.LogMessage("Duplicate candidate '{0}' found in ResolvedFilesToPublish", candidate.ItemSpec);
+                        }
                         continue;
                     }
 
-                    resolvedAssemblyToPublish.Add(Path.GetFileName(candidate.GetMetadata("RelativePath")), candidate);
+                    var candidateName = Path.GetFileName(candidate.GetMetadata("RelativePath"));
+                    if (!resolvedAssemblyToPublish.ContainsKey(candidateName))
+                    {
+                        resolvedAssemblyToPublish.Add(candidateName, candidate);
+                    }
+                    else
+                    {
+                        Log.LogMessage("Duplicate candidate '{0}' found in ResolvedFilesToPublish", candidate.ItemSpec);
+                    }
+
                     continue;
                 }
-                if (string.Equals(extension, " .pdb", StringComparison.Ordinal))
+
+                if (string.Equals(extension, ".pdb", StringComparison.Ordinal))
```

Comment:
Was the extra space before `.pdb` a bug or intentional?

Reply:
It was an ad-hoc bug fix